### PR TITLE
EVG-3137 Fix task group ordering in queue

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: scheduler/setup_funcs.go [hunk 1]
```diff
@@ -53,3 +56,21 @@ func cacheSimilarFailing(comparator *CmpBasedTaskComparator) (err error) {
...
 }
+
+// groupTaskGroups puts tasks that have the same build and task group next to
+// each other in the queue. This ensures that, in a stable sort,
+// byTaskGroupOrder sorts task group members relative to each other.
+func groupTaskGroups(comparator *CmpBasedTaskComparator) error {
+	taskMap := make(map[string]task.Task)
+	taskKeys := []string{}
+	for _, t := range comparator.tasks {
+		k := fmt.Sprintf("%s_%s_%s", t.BuildId, t.TaskGroup, t.Id)
```
Comment (tychoish) [inline scheduler/setup_funcs.go]: if I asked really nicely could you use hyphens or dots rather than underscores?

File: scheduler/setup_funcs.go [hunk 2]
```diff
@@ -53,3 +56,21 @@ func cacheSimilarFailing(comparator *CmpBasedTaskComparator) (err error) {
...
+		taskMap[k] = t
+		taskKeys = append(taskKeys, k)
+	}
+	sort.Strings(taskKeys)
```
Comment (tychoish) [inline scheduler/setup_funcs.go]: does this also need to be stable?

File: scheduler/setup_funcs.go (hunk 1 above)
Comment (bsamek) [reply scheduler/setup_funcs.go]: lol sure

File: scheduler/setup_funcs.go (hunk 2 above)
Comment (bsamek) [reply scheduler/setup_funcs.go]: No. This sort has only one possible end state, since task ID is part of the key.

**Pull request merged**